activerecord/test/cases/fixtures_test.rb

```diff
@@ -1482,6 +1482,14 @@ def test_only_existing_connections_are_replaced
       end
     end
 
+    def test_connections_with_no_writing_role_are_skipped
+      AbstractCompany.connects_to(database: { reading: :readonly })
```

Comment:
So Rails kind of requires that there's a writing role, at least how it's configured currently. If you don't include one and boot a console, and try to query the db, you'll get an error because the default is to lookup on the writing role. I think this will be easier to change with granular connection management, but I haven't gotten around to doing that. Description of the problem is here https://github.com/rails/rails/issues/39580

Reply:
Thanks for the context, I hadn't seen https://github.com/rails/rails/issues/39580.

My actual goal here is https://github.com/rails/rails/pull/40384, so that I can write tests for some code in our application that switches connections manually. Since that requires pool sharing to be reversible, I'm trying to understand all of the edge cases so that I'm confident they'll reverse correctly.

It's possible to write production code that works fine with no writing connection, as long as you wrap every query in a `connected_to` block, so I thought it was worthwhile to make that code behave the same during transactional tests. If you'd prefer not to add code to handle this case though, since it's not really supported, I'm fine with ignoring it.